### PR TITLE
Add Trino 421 release notes

### LINKED ISSUE #18021: Release notes for 421

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 421 release. Fixes #18021 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 22 June 2023

* #18009 ✅ rn ✅ docs

## 23 June 2023

* #17623 ✅ rn ✅ docs
* #17869 ✅ rn ✅ docs

## 24 June 2023

* #18007 ✅ rn ✅ docs
* #17342 ✅ rn ✅ docs

## 25 June 2023

* #18015 ❌ rn ✅ docs
* #18037 ✅ rn ✅ docs
* #17995 ✅ rn ✅ docs

## 26 June 2023

* #18008 ✅ rn ✅ docs
* #18043 ✅ rn ✅ docs
* #16308 ✅ rn ✅ docs
* #18034 ✅ rn ✅ docs
* #18039 ✅ rn ✅ docs
* #18048 ✅ rn ✅ docs
* #18049 ✅ rn ✅ docs

## 27 June 2023

* #18042 ✅ rn ✅ docs
* #18044 ✅ rn ✅ docs
* #17988 ✅ rn ✅ docs
* #18058 ✅ rn ✅ docs
* #18059 ✅ rn ✅ docs
* #18029 ✅ rn ✅ docs
* #18061 ✅ rn ✅ docs
* #18051 ✅ rn ✅ docs
* #17984 ✅ rn ✅ docs

## 28 June 2023

* #17900 ✅ rn ✅ docs
* #18024 ✅ rn ✅ docs
* #18023 ✅ rn ✅ docs
* #18079 ✅ rn ✅ docs
* #18076 ✅ rn ✅ docs
* #18035 ✅ rn ✅ docs

## 29 June 2023

* #18083 ✅ rn ✅ docs
* #18086 ✅ rn ✅ docs
* #18077 ✅ rn ✅ docs
* #15026 ✅ rn ✅ docs
* #18093 ✅ rn ✅ docs

## 30 June 2023

* #18080 ✅ rn ✅ docs
* #18094 ✅ rn ✅ docs
* #17816 ✅ rn ✅ docs
* #18103 ✅ rn ✅ docs

## 1 July 2023

* #18087 ✅ rn ✅ docs
* #16191 ✅ rn ✅ docs

## 2 July 2023

* #18010 ✅ rn ✅ docs

## 3 July 2023

* #18006 ✅ rn ✅ docs
* #17999 ✅ rn ✅ docs
* #18018 ✅ rn ✅ docs
* #17394 ✅ rn ✅ docs
* #17372 ✅ rn ✅ docs
* #17195 ✅ rn ✅ docs
* #17710 ✅ rn ✅ docs

## 4 July 2023

* #18085 ✅ rn ✅ docs
* #18114 ✅ rn ✅ docs
* #18081 ✅ rn ✅ docs
* #18128 ✅ rn ✅ docs
* #18131 ✅ rn ✅ docs

## 5 July 2023

* #18131 ✅ rn ✅ docs
* #18125 ✅ rn ✅ docs
* #18109 ✅ rn ✅ docs
* #17565 ✅ rn ✅ docs
* #17939 ✅ rn ❌ docs
* #16280 ❌ rn ❌ docs
* #18108 ✅ rn ✅ docs
* #18092 ✅ rn ✅ docs
* #17563 ✅ rn ✅ docs
* #17050 ✅ rn ✅ docs
* #18078 ✅ rn ✅ docs
* #18142 ✅ rn ✅ docs
* #17919 ✅ rn ✅ docs
* #17527 ✅ rn ✅ docs

## 6 July 2023

* #17979 ✅ rn ✅ docs
* #18047 ✅ rn ✅ docs
* #18144 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-421.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Release 421 (28 Jun 2023)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-421.md:1]: 5 Jul

File: docs/src/main/sphinx/release/release-421.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+* Add support for {doc}`/sql/insert` statements. ({issue}`3866`)
+
+## Hive connector
+
+* Add Hive partition projection column properties to the output of
+  `SHOW CREATE TABLE`. ({issue}`18076`)
+* Fix incorrect query results when using S3 Select with `IS NULL` or
+  `IS NOT NULL` predicates. ({issue}`17563`)
+* Fix incorrect query results when using S3 Select and a table's null format is
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-421.md:35]: whats a "table's null format" ?

File: docs/src/main/sphinx/release/release-421.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Kafka connector
+
+* Add support for reading Protobuf messages containing the `Any` Protobuf type.
+  This is disabled by default and can be enabled by setting the 
+  `kafka.protobuf-any-support-enabled` configuration property to `true`. ({issue}`17394`)
+
+## MongoDB connector
+
+* Improve query performance on tables with `row` (struct) columns when only a
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-421.md:51]: can we leave "struct" out?

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by mosabua — APPROVED** [review]
Looks good now. We have to fast follow on docs after the release.

File: docs/src/main/sphinx/release/release-421.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Release 421 (5 Jul 2023)
+
+## General
+
+* Add support for check constraints in an `UPDATE` statement. ({issue}`17195`)
+* Improve performance for queries involving a `year` function within an `IN`
+  predicate. ({issue}`18092`)
+* Fix query failure when cancelling a query with a window function. ({issue}`18061`)
+* Fix query failure for queries involving the `concat_ws` function on arrays
```
Comment (martint) [inline docs/src/main/sphinx/release/release-421.md:9]: Fix failure for queries ...

File: docs/src/main/sphinx/release/release-421.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
+# Release 421 (5 Jul 2023)
+
+## General
+
+* Add support for check constraints in an `UPDATE` statement. ({issue}`17195`)
+* Improve performance for queries involving a `year` function within an `IN`
+  predicate. ({issue}`18092`)
+* Fix query failure when cancelling a query with a window function. ({issue}`18061`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-421.md:8]: Fix failure when cancelling ...

File: docs/src/main/sphinx/release/release-421.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+* Fix query failure for queries involving the `concat_ws` function on arrays
+  with more than 254 values. ({issue}`17816`)
+* Fix loss of timestamp precision when coercing a
```
Comment (martint) [inline docs/src/main/sphinx/release/release-421.md:11]: This doesn't seem to match what the PR is about. What's the observed behavior before that PR? A failure? Correctness issues? cc @Praveen2112

File: docs/src/main/sphinx/release/release-421.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  [structural data type](structural-data-types) that contains a timestamp. ({issue}`17900`)
+
+## JDBC driver
+
+* Add support for using an alternative hostname for full SSL verification
```
Comment (martint) [inline docs/src/main/sphinx/release/release-421.md:16]: "when SSLVerification is set to FULL"

**Pull request merged**